pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -70,6 +70,13 @@ func (p *ldapProvider) loginUser(lConn ldapv3.Client, credential *v32.BasicLogin
 		return v3.Principal{}, nil, httperror.WrapAPIError(err, httperror.ServerError, "server error while authenticating")
 	}
 
+	if config.SearchUsingServiceAccount {
```

Comment:
Should we guard against doing the bind twice?

`ldap.AuthenticateServiceAccountUser` does a Bind and we're already binding here https://github.com/rancher/rancher/blob/9f4406e0427231c1437e4d441c1ace97c5a9b205/pkg/auth/providers/ldap/ldap_client.go#L65

A Bind involves a network operation.

Reply:
Maybe I'm misunderstanding what you're thinking about here.  If this option is enabled, I think we wind-up needing to bind again.....unless we did some refactoring to maybe have multiple connections (each would only be bound once?).
Definitely open to alternate approaches that are more efficient.